### ISSUE TITLE
Easy way to restart OVS services

### ISSUE THREAD

**Issue by wimpers** [openvstorage/framework#719]
Currently there is no easy way to restart all OVS services on a node. A command line might be useful. 
We can later integrate this in the support portal if needed.

**Comment by pploegaert** [comment]
there is a quick hack ovs stop services, but you might have to run it twice ... to be improved ...

**Comment by wimpers** [comment]
This one is for @kvanhijf ;-)

**Comment by khenderick** [comment]
@wimpers, what features do you want? Only the framework services or truly all services on a given node (e.g. volumedriver, dtls, proxies, asds, ...). Restarting all ovs services on a given node might be risky for e.g. running volumes or backend availability.

Restarting only framework services is already possible by just restarting `ovs-watcher-framework` (or starting or stopping it for that matter).

**Comment by wimpers** [comment]
I've created this one as @kvanhijf had to write a script to stop (all?) services. The reason I created this bug FR is to tackle this kind of scenario's where  people have to write their own script. 

@kvanhijf can you list the services which you had to stop?

**Comment by khenderick** [comment]
He wrote a script to stop `ovs-watcher-framework` on all nodes. We can write such shortcut (e.g. `ovs stop framework` or whatever. The important question is whether you want to stop _all_ services or only the framework services.

**Comment by khenderick** [comment]
After discussion with @wimpers:
- ovs start/stop services: remove
- ovs start/stop/restart framework: alias for start/stop/restart ovs-watcher-framework
- ovs start/stop/restart framework _ip_: see above, on the given node
- ovs start/stop/restart framework all: see above, on all nodes (requires framework to be up-and-running)

For now, no alternatives for other services (arakoons, etcds, volumedrivers, ...) as they have more impact and should thus be executed manually and carefully.

**Comment by wimpers** [comment]
After discussion with @khenderick 

Restarting (proxies, volumedrivers, arakoons, etcd, ...) is tricky so we will start with the basics

```
ovs stop framework = stop ovs-watcher-service
ovs stop framework <ip> = stop ovs-watcher-service on that node
ovs stop framework all = stop ovs-watcher-service op all nodes
```

Same goes for restart and start of course.

Full cli will be picked up as part of https://github.com/openvstorage/home/issues/28

**Comment by khenderick** [comment]
Fixed by #1004, packaged in openvstorage-2.7.4-rev.4092.7a6ff12

**Comment by JeffreyDevloo** [comment]
### Steps
- Installed latest packages
- Installed cluster
- Added roles to disks
- Added backend
- Initialized disks and claimed asds
- Added usable preset
- Added vpool

Testing the commands and verifying using the GUI:
- Executed 'ovs help' to get the entry
- Executed 'ovs stop framework'
- Verified that ovs was down (checking ovs-watcher-framework and gui)
- ...

### Output

**ovs stop framework:**
_Checking service:_

```
root@ovs-node1:~# ovs stop framework
Stopping...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
```

_Browsing to 10.100.199.151:_
Got redirected to a different node that is online.

_Check status of the node via GUI:_
Got warning mark within 30s:
![selection_113](https://cloud.githubusercontent.com/assets/17570109/19438972/9348b624-947c-11e6-982b-6046cf94ccb6.png)
Got failure mark within 120s:
![selection_114](https://cloud.githubusercontent.com/assets/17570109/19439034/e34e2dc0-947c-11e6-90e2-a4b1f45dd145.png)

**ovs start framework:**
_Checking service:_

```
root@ovs-node1:~# ovs start framework
Starting...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework start/running, process 3710
root@ovs-node1:~# 
```

_Browsing to 10.100.199.151:_
page accessible
_Check status of the node via GUI:_
Got the OK mark within 5s:
![selection_115](https://cloud.githubusercontent.com/assets/17570109/19439178/80498f2a-947d-11e6-9a39-3b3801b867d4.png)

**ovs stop framework 10.100.199.151:**
_Checking service:_

```
root@ovs-node1:~# ovs stop framework 10.100.199.151
Stopping on ovs-node1...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
root@ovs-node1:~# 
```

_Browsing to 10.100.199.151:_
Got redirected to another node that is online.
_Checking status of the node via GUI:_
Got warning mark within 30s:
![selection_113](https://cloud.githubusercontent.com/assets/17570109/19438972/9348b624-947c-11e6-982b-6046cf94ccb6.png)
Got failure mark within 120s:
![selection_114](https://cloud.githubusercontent.com/assets/17570109/19439034/e34e2dc0-947c-11e6-90e2-a4b1f45dd145.png)

**ovs start framework 10.100.199.151:**
_Checking service:_

```
root@ovs-node1:~# ovs start framework 10.100.199.151
Starting on ovs-node1...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework start/running, process 4851
root@ovs-node1:~# 

```

_Browsing to 10.100.199.151:_
Page accessible.
_Check status of the node via GUI:_
Got the OK mark within 5s:
![selection_115](https://cloud.githubusercontent.com/assets/17570109/19439178/80498f2a-947d-11e6-9a39-3b3801b867d4.png)

ovs stop framework all:
_Checking service:_

```
root@ovs-node1:~# ovs stop framework all
Stopping on ovs-node2...
Stopping on ovs-node3...
Stopping on ovs-node1...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
root@ovs-node2:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
root@ovs-node3:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
```

_Browsing to any IP:_
No redirection (all nodes are down).
ovs start framework all:
_Checking service:_

```
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework stop/waiting
root@ovs-node1:~# ovs start framework all
Starting on ovs-node2...
Starting on ovs-node3...
Starting on ovs-node1...
Done
root@ovs-node1:~# service ovs-watcher-framework status
ovs-watcher-framework start/running, process 5527
root@ovs-node2:~# service ovs-watcher-framework status
ovs-watcher-framework start/running, process 26324
root@ovs-node3:~# service ovs-watcher-framework status
ovs-watcher-framework start/running, process 25791
```

_Browsing to all IP:_
All pages are accessible.
_Check status of the node via GUI:_
All nodes are marked as ok.
![selection_116](https://cloud.githubusercontent.com/assets/17570109/19439594/e5c1009e-947e-11e6-89ca-da3045f9fd74.png)

### Test result

Test passed.

### Packages:
- openvstorage                       2.7.4-rev.4159.0707c77-1                                 amd64        openvStorage
- openvstorage-backend               1.7.4-rev.766.0d9f30e-1                                  amd64        openvStorage Backend plugin
- openvstorage-backend-core          1.7.4-rev.766.0d9f30e-1                                  amd64        openvStorage Backend plugin core
- openvstorage-backend-webapps       1.7.4-rev.766.0d9f30e-1                                  amd64        openvStorage Backend plugin Web Applications
- openvstorage-cinder-plugin         1.2.2-rev.38.dcc3b76-1                                   amd64        OpenvStorage Cinder plugin for OpenStack
- openvstorage-core                  2.7.4-rev.4159.0707c77-1                                 amd64        openvStorage core
- openvstorage-hc                    1.7.4-rev.766.0d9f30e-1                                  amd64        openvStorage Backend plugin HyperConverged
- openvstorage-sdm                   1.6.4-rev.388.1f4375d-1                                  amd64        Open vStorage Backend ASD Manager
- openvstorage-webapps               2.7.4-rev.4159.0707c77-1                                 amd64        openvStorage Web Applications
